### ISSUE TITLE
IPIP traffic not properly handled when using "ipip" in secgroups

### ISSUE THREAD

**Issue by mnaser** [kubernetes-sigs/cluster-api-provider-openstack#1484]
/kind bug

**What steps did you take and what happened:**
We're seeing an issue that we're troubleshooting where certain environments seem like they are happier using '4' instead of 'ipip' in the security group.   Causing IPIP traffic for Calico not to flow properly

https://github.com/vexxhost/magnum-cluster-api/issues/72

**What did you expect to happen:**
IPIP traffic to flow with no issues

**Anything else you would like to add:**
This seems to have been started by https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1290

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built):
- Cluster-API version:
- OpenStack version:
- Minikube/KIND version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by jds9090** [comment]
Check your Openstack supports it. I think It depends on the version of your Openstack.

**Comment by mnaser** [comment]
@jds9090 i looked into the code and it seems like there might be a bug in Neutron that is causing this, but perhaps if we keep using 4 then we’ll know even older environments will work

**Comment by jichenjc** [comment]
I did the original 4=> ipip to make it clearer
I am not against the desire to revert the PR if it break some function

**Comment by jichenjc** [comment]
@mnaser can you help provide some detail about `neutron bug` you mentioned so we can take that as reference
in the code then revert back to `4`

**Comment by mnaser** [comment]
ok, I did some checks: @jichenjc.. when you use proto `ipip`, it uses proto. number `94`, this is because:

```
root@abb2f5652d8f:/# egrep '^(ipencap|ipip)' /etc/protocols
ipencap	4	IP-ENCAP	# IP encapsulated in IP (officially ``IP'')
ipip	94	IPIP		# IP-within-IP Encapsulation Protocol
```

So since Neutron makes sure to check what are the supported values of the local system first, it'll end up binding `ipip` to `94`.  I can also see that in my iptables rule tests on the compute hosts when changing it.

Unfortunately, the API doesn't accept `ipencap` and I'm not really sure what is the right protocol number based on the standard, but what we know is calico wants ip number 4, and neutron will give you 94 if you request `ipip` cause thats what the OS says `ipip` is equal to.

**Comment by jds9090** [comment]
@mnaser @jichenjc Thanks for sharing your findings. I am experiencing the same situation as follows.
```
root@xxx:ubuntu:> egrep '^(ipencap|ipip)' /etc/protocols
ipencap	4	IP-ENCAP	# IP encapsulated in IP (officially ``IP'')
ipip	94	IPIP		# IP-within-IP Encapsulation Protocol
```

so, reasonable to revert back to 4

**Comment by mnaser** [comment]
> @mnaser @jichenjc Thanks for sharing your findings. I am experiencing the same situation as follows.
> 
> ```
> 
> root@xxx:ubuntu:> egrep '^(ipencap|ipip)' /etc/protocols
> 
> ipencap	4	IP-ENCAP	# IP encapsulated in IP (officially ``IP'')
> 
> ipip	94	IPIP		# IP-within-IP Encapsulation Protocol
> 
> ```
> 
> 
> 
> so, reasonable to revert back to 4

This should be fixed in the hopefully upcoming 0.7.1 release :)
